### ISSUE TITLE
Verify Privileged Groups contains only desired members

### ISSUE THREAD

**Issue by PrzemyslawKlys** [EvotecIT/Testimo#24]

**Comment by MarcLaf** [comment]
Chadmcox has a great script here https://github.com/chadmcox/ADPoSh/blob/master/Privileged%20Objects/FindandFixADObjectswithStaleAdminSDHolder.ps1
This checks all users with Admin Count 1 and then cross references with the group memberships and shows the difference. Only thing is his script requires admin privs for some reason. If you test it use -reportonly so it doesn't automatically modify attributes.

**Comment by PrzemyslawKlys** [comment]
I've my own version (I've seen Chadmcox) - I've already checked for OrphanedObjects - https://github.com/EvotecIT/Testimo/blob/master/Private/SourcesForest/OrphanedAdmins.ps1 

The test is based on this: https://github.com/EvotecIT/ADEssentials/blob/master/Public/Get-WinADPriviligedObjects.ps1 it has multiple switches including getting legitimate AdminCount users. 

I've not added this yet as the thing is I am not able to tell True/False for this test. I need to expand Testimo with a new type of tests where I will just gather data and display it for verification.

**Comment by MarcLaf** [comment]
Ah okay. Yeah I saw the check in the script but the results made me a bit confused because it was showing all and not just the incorrect ones.

**Comment by PrzemyslawKlys** [comment]
hrmms, it should show only incorrect ones. Does it show more?

**Comment by MarcLaf** [comment]
I seem to recall it showing ALL accounts with Admin Count 1 regardless of them belonging to a built in privileged group. That's what prompted me to explore it more and stumble upon Chad's script.

**Comment by PrzemyslawKlys** [comment]
Well, what it does it checks Orphaned Admin Count. This means it lists all those users with Admin Count set to 1 where they don't actually have any admin rights. It's potentially risky. However, that command has also other switches like listing proper admins (groups/users/computers).

**Comment by MarcLaf** [comment]
Yeah that's what I thought it was supposed to do but the results that came back showed all accounts whether they really belonged to a priveleged group or not. It wasn't only showing orphaned.

**Comment by PrzemyslawKlys** [comment]
Ok, that's weird. I'll review. 

It should:
- show users/groups/computers that admin count is 1 and it shouldn't be only.

**Comment by MarcLaf** [comment]
Oh I'm sorry I misunderstood. I thought it should only show me accounts with Admin Count set to 1 that do not belong to a privileged group. Yes it is showing all accounts where Admin Count is set to 1.
I'm running the report again, just waiting for results to double check.

**Comment by PrzemyslawKlys** [comment]
Yes, that's what it should show you. 

- An account (user, computer) or a group that used to be an Admin, but now isn't. If it's showing more then that... I need to review why.

**Comment by MarcLaf** [comment]
Okay understood. Thank you so much for this! I've learned a bunch more ever since going over the results these past few days. Awesome work 👍

**Comment by MarcLaf** [comment]
By the way, results came back and yes it's showing all accounts with AdminCount set to 1, not just orphaned.

**Comment by PrzemyslawKlys** [comment]
Can you run those two commands? 

- Get-WinADPriviligedObjects | ft -AutoSize
- Get-WinADPriviligedObjects -OrphanedOnly | ft -AutoSize

First one should display all members with AdminCount, but GroupDomain GroupDistinguishedName will be only filled in for "proper" admins. When empty it means it's orphaned admin group/user/computer. 

2nd one should show only orphaned admins.
